### ISSUE TITLE
Recoil::all() does not handle empty arrays

### ISSUE THREAD

**Issue by tsufeki** [recoilphp/kernel#3]
Using `yield [];` in a coroutine results in
```
Recoil\Kernel\Exception\KernelStoppedException: The kernel was stopped before the main strand exited.
```
and its traceback doesn't point to the code that caused this.

I think `yield [];` should be equivalent to `yield;` or return immediately or, alternatively, a mention in the docs be made.

(I'm using React kernel.)

Thank you for your work.

**Comment by jmalloc** [comment]
Thanks for the report! I'll investigate this as soon as possible. My instinct is that this should simply return immediately as you suggest.

**Comment by jmalloc** [comment]
Changes are available in `recoil/kernel` version 1.0.1, which should be pulled into your project via `recoil/react`.  

I opted to make `yield [];` equivalent to `yield;` as per your first suggestion, so that the `all()` operation is consistently cooperative.

I have also attempted to provide consistent semantics for the other composite operations: `any()`, `some()` and `first()`.

**Comment by tsufeki** [comment]
Great, thank you.
